src/PowerShell/Microsoft.WinGet.Client.Engine/Commands/Common/FinderCommand.cs

```diff
@@ -35,31 +35,33 @@ internal FinderCommand(PSCmdlet psCmdlet)
         /// Gets or sets the field that is matched against the identifier of a package.
         /// </summary>
         [Filter(Field = PackageMatchField.Id)]
-        protected string Id { get; set; }
+        protected string? Id { get; set; }
 
         /// <summary>
         /// Gets or sets the field that is matched against the name of a package.
         /// </summary>
         [Filter(Field = PackageMatchField.Name)]
-        protected string Name { get; set; }
+        protected string? Name { get; set; }
 
         /// <summary>
         /// Gets or sets the field that is matched against the moniker of a package.
         /// </summary>
         [Filter(Field = PackageMatchField.Moniker)]
-        protected string Moniker { get; set; }
+        protected string? Moniker { get; set; }
 
         /// <summary>
         /// Gets or sets the name of the source to search for packages. If null, then all sources are searched.
         /// </summary>
-        protected string Source { get; set; }
+        protected string? Source { get; set; }
 
         /// <summary>
         /// Gets or sets how to match against package fields.
         /// </summary>
-        protected string[] Query { get; set; }
+#pragma warning disable SA1011 // Closing square brackets should be spaced correctly
+        protected string[]? Query { get; set; }

```

Comment:
Academic question: Does `string?[]` mean a non-null array of nullable strings, while `string[]?` means a nullable array of non-null strings?  If so, then `string?[]?` is a lovingly confusing type name 😄

Reply:
Yup. There's another place where I needed a non-nullable array with nullable objects, so it was object?[]